src/components/settings/ImportAllDialog/index.tsx

```diff
@@ -0,0 +1,227 @@
+import { useAppDispatch } from '@/store'
+
+import ErrorMessage from '@/components/tx/ErrorMessage'
+import { useDropzone } from 'react-dropzone'
+import { addedSafesSlice } from '@/store/addedSafesSlice'
+import { addressBookSlice } from '@/store/addressBookSlice'
+
+import css from './styles.module.css'
+import Box from '@mui/material/Box'
+import Grid from '@mui/material/Grid'
+import IconButton from '@mui/material/IconButton'
+import type { MouseEventHandler } from 'react'
+import { useGlobalImportJsonParser } from './useGlobalImportFileParser'
+import { showNotification } from '@/store/notificationsSlice'
+import { Alert, AlertTitle, Link, SvgIcon } from '@mui/material'
+import FileIcon from '@/public/images/settings/data/file.svg'
+import { SETTINGS_EVENTS, trackEvent } from '@/services/analytics'
+
+const ImportAllDialog = ({ handleClose }: { handleClose: () => void }): ReactElement => {
+  const [jsonData, setJsonData] = useState<string>()
+  const [fileName, setFileName] = useState<string>()
+  const [error, setError] = useState<string>()
+
+  // Parse the jsonData whenever it changes
+  const { addedSafes, addedSafesCount, addressBook, addressBookEntriesCount } = useGlobalImportJsonParser(jsonData)
+
+  const dispatch = useAppDispatch()
+
+  const onDrop = (acceptedFiles: File[]) => {
+    if (acceptedFiles.length === 0) {
+      return
+    }
+    const file = acceptedFiles[0]
+    const reader = new FileReader()
+    reader.onload = function (evt) {
+      if (!evt.target) {
+        return
+      }
+      if (typeof evt.target.result !== 'string') {
+        return
+      }
+      setJsonData(evt.target.result)
+      setFileName(file.name)
+    }
+    reader.readAsText(file)
+  }
+
+  const onRemove: MouseEventHandler = (event) => {
+    setJsonData(undefined)
+    setFileName(undefined)
+    event.preventDefault()
+    event.stopPropagation()
+  }
+
+  const { getRootProps, getInputProps, isDragActive, isDragAccept, isDragReject } = useDropzone({
+    maxFiles: 1,
+    onDrop,
+    accept: {
+      text: ['.json'],
+    },
+  })
+
+  const handleImport = () => {
+    if (!jsonData) {
+      return
+    }
+
+    if (addressBook) {
+      dispatch(addressBookSlice.actions.setAddressBook(addressBook))
+    }
+
+    if (addedSafes) {
+      dispatch(addedSafesSlice.actions.setAddedSafes(addedSafes))
+    }
+
+    dispatch(
+      showNotification({
+        variant: 'success',
+        groupKey: 'global-import-success',
+        message: 'Successfully imported data',
+        detailedMessage: `${addedSafesCount > 0 ? `${addedSafesCount} Safes were added. \n` : ''}${
+          addressBookEntriesCount > 0 ? `${addressBookEntriesCount} addresses were added to your address book.` : ''
+        }`,
+      }),
+    )
+
+    trackEvent({
+      ...SETTINGS_EVENTS.DATA.IMPORT_ALL,
+      label: `Added Safes: ${addedSafesCount}, Addressbook entries: ${addressBookEntriesCount}`,
+    })
```

Comment:
I would split these into two events. It's not possible to filter the data well otherwise.